src/server/game/Scripting/ScriptMgr.cpp

```diff
@@ -408,6 +408,11 @@ void ScriptMgr::OnAfterConfigLoad(bool reload)
     FOREACH_SCRIPT(WorldScript)->OnAfterConfigLoad(reload);
 }
 
+void ScriptMgr::OnBeforeFinalizePlayerWorldSession(uint32& cacheVersion)
```

Comment:
For primitive data types like in this case int you should just pass them by value there's no need to pass by reference here.

Reply:
Normally I would agree but in this specific case pass by reference is required as far as I'm aware as I want to be able to modify what the cacheVersion actually is that gets sent to clients.